knowledge_graph/classifier/keyword.py

```diff
@@ -51,11 +51,41 @@ def __init__(self, concept: Concept):
         """
         super().__init__(concept)
 
+        def make_separator_flexible(label: str) -> str:
+            r"""
+            Convert a label to a regex pattern that matches different word separators.
+
+            This allows labels like "greenhouse gas" to match:
+            - "greenhouse gas" (space)
+            - "greenhouse-gas" (hyphen)
+            - "greenhouse\ngas" (newline)
+            - "greenhouse -gas" (multiple consecutive separators)
+
+            :param str label: The label to convert
+            :return str: A regex pattern string that matches the label with flexible separators
+            """
+            # Split by any common separator characters (space, hyphen, newline, etc.)
+            separator_pattern = r"[\s\-]+"
```

Comment:
do we want to split on both `–` and `-`? I know only the latter is actually correct, but I imagine we could get some misplaced em-dashes from our parser.

Reply:
very good point! will add hyphens, en dashes, and em dashes explicitly